performance/include/traccc/resolution/fitting_performance_writer.hpp

```diff
@@ -103,7 +112,9 @@ class fitting_performance_writer {
         if (fit_res.ndf > 0 && !trk_state.is_hole) {
             write_res(truth_param, trk_state.smoothed(), ptc);
         }
-        write_stat(fit_res, track_states_per_track);
+        if (fit_res.fit_outcome == fitter_outcome::SUCCESS) {
```

Comment:
I guess this check is technically not necessary because you return if the outcome was not a success on lines 61-63.